src/api/services/process/processService.ts

```diff
@@ -0,0 +1,87 @@
+import type { Config } from "../../config/type";
+import { type Grade, Grades } from "../../type";
+import type { Member, MemberInfo, Submission } from "../common/types";
+
+export function createProcessService(config: Config) {
+  return {
+    analyzeMemberInfo(
+      members: Member[],
+      submissions: Submission[],
+    ): MemberInfo[] {
+      const memberMap = initializeMemberMap(members);
+      updateSubmissions(memberMap, submissions);
+      calculateProgress(memberMap, config.study.totalProblemCount);
+      updateGrades(memberMap, config.study.gradeThresholds);
+
+      return Object.values(memberMap);
+    },
+  };
+}
+
+const initializeMemberMap = (members: Member[]): Record<string, MemberInfo> => {
+  const memberMap: Record<string, MemberInfo> = {};
+
+  members.forEach((member) => {
+    memberMap[member.id] = {
+      ...member,
+      totalSubmissions: 0,
+      submissions: [],
+      progress: 0,
+      grade: Grades.SEED,
```

Comment:
Enum을 쓰신 것도 아니고, 키와 값이 동일한 객체를 이런 식으로 이용하는 것은 상당히 부자연스럽게 느껴지네요. `grade` 속성이 이미 `Grade` 타입이기 때문에 그냥 문자열을 할당하는 것이 더 일반적인 코딩 패턴일 것입니다. 범위에서 벗어나는 문자열을 할당하면 타입 오류가 발생하기 때문에 그냥 문자열을 할당해도 충분히 type-safe합니다.

```suggestion
      grade: "SEED",
```